docs/adr/0001_cmsis_component_source_structure_recommendations.md

```diff
@@ -0,0 +1,119 @@
+# CMSIS component source structure recommendations
+Date: 2020-08-19
+
+## Status
+Proposed
+
+## Context
+
+[`ARM-software/CMSIS_5`](https://github.com/ARM-software/CMSIS_5) has been ported to Mbed OS. However, it is split across multiple directories and it is not immediately obvious that it comes from that directory. It also makes it difficult to create an independent CMake target can be built as a library that can be optionally be added by end users.
+
+The current proposal is to create a new directory `CMSIS_5/` at the root of the repository that will mirror closely the structure used by `ARM-software/CMSIS_5/`. It will result in the removal of `rtos/source/TARGET_CORTEX/` and `cmsis/`.
```

Comment:
Could you detail what _mirror closely_ means ? Is it just related to the `TARGET_` folders to accommodate mbed-cli or is there other changes made to the `ARM-software/CMSIS_5` source tree ?

Reply:
[Done](https://github.com/ARMmbed/mbed-os/compare/9859c6c228e89f9e6ba52f97cb0070f6b7c6ed71..f6d9bb1e8ba359bd2c07560a6f8122adeccf513a).

Reply:
Thanks @hugueskamba . What is the plan moving forward regards to pulling CMSIS 5 ? If we want to turn it into a component then how do we handle required modifications ?

Reply:
I believe the answer to that question is README: https://github.com/ARMmbed/mbed-os/pull/13464/files#diff-0ffd2fb9a2181108e2505172559c939f

Reply:
That's interesting! 
@0xc0170 How do we define tools that have to be run when a package is imported with CMSIS pack ? Is it handled by the [generator](https://www.keil.com/pack/doc/CMSIS/Pack/html/pdsc_generators_pg.html) element ? It is unclear when generators are run in a package life cycle.

Reply:
> @0xc0170 How do we define tools that have to be run when a package is imported with CMSIS pack ? Is it handled by the generator element ? It is unclear when generators are run in a package life cycle.

I don't know yet, haven't used this generator . I only read  about it before.